stl/inc/spanstream

```diff
@@ -0,0 +1,386 @@
+    using char_type   = _Elem;
+    using int_type    = typename _Traits::int_type;
+    using pos_type    = typename _Traits::pos_type;
+    using off_type    = typename _Traits::off_type;
+    using traits_type = _Traits;
+
+    // N4892 [ospanstream.ctor], constructors
+    explicit basic_ospanstream(_STD span<_Elem> _Span, ios_base::openmode _Which = ios_base::out)
+        : _Mybase(_STD addressof(_Buf)), _Buf(_Span, _Which | ios_base::out) {}
+
+    basic_ospanstream(const basic_ospanstream&) = delete;
+
+    basic_ospanstream(basic_ospanstream&& _Right) : _Mybase(_STD move(_Right)), _Buf(_STD move(_Right._Buf)) {
+        _Mybase::set_rdbuf(_STD addressof(_Buf));
+    }
+
+    // N4892 [ospanstream.assign], assignment and swap
+    basic_ospanstream& operator=(const basic_ospanstream&) = delete;
+
+    basic_ospanstream& operator=(basic_ospanstream&& _Right) {
+        _Mybase::swap(_Right);
+        _STD swap(_Buf, _Right._Buf);
+    }
+
+    void swap(basic_ospanstream& _Right) {
+        _Mybase::swap(_Right);
+        _STD swap(_Buf, _Right._Buf);
+    }
+
+    // N4892 [ospanstream.members], member functions
+    _Mysb* rdbuf() const noexcept {
+        return const_cast<_Mysb*>(_STD addressof(_Buf));
+    }
+
+    span<const _Elem> span() const noexcept {
+        return this->rdbuf()->span();
+    }
+
+    void span(_STD span<_Elem> _Span) noexcept {
+        this->rdbuf()->span(_Span);
+    }
+
+private:
+    _Mysb _Buf;
+};
+
+template <class _Elem, class _Traits>
+void swap(basic_ospanstream<_Elem, _Traits>& _Left, basic_ospanstream<_Elem, _Traits>& _Right) {
+    _Left.swap(_Right);
+}
+
+template <class _Elem, class _Traits>
+class basic_spanstream : public basic_iostream<_Elem, _Traits> {
+public:
+    using _Mybase     = basic_iostream<_Elem, _Traits>;
+    using _Mysb       = basic_spanbuf<_Elem, _Traits>;
+    using char_type   = _Elem;
+    using int_type    = typename _Traits::int_type;
+    using pos_type    = typename _Traits::pos_type;
+    using off_type    = typename _Traits::off_type;
+    using traits_type = _Traits;
+
+    // N4892 [spanstream.ctor], constructors
+    explicit basic_spanstream(_STD span<_Elem> _Span, ios_base::openmode _Which = ios_base::out | ios_base::in)
+        : _Mybase(_STD addressof(_Buf)), _Buf(_Span, _Which) {}
+
+    basic_spanstream(const basic_spanstream&) = delete;
+
+    basic_spanstream(basic_spanstream&& _Right) : _Mybase(_STD move(_Right)), _Buf(_STD move(_Right._Buf)) {
+        _Mybase::set_rdbuf(_STD addressof(_Buf));
+    }
+
+    // N4892 [spanstream.assign], assignment and swap
+    basic_spanstream& operator=(const basic_spanstream&) = delete;
+
+    basic_spanstream& operator=(basic_spanstream&& _Right) {
+        _Mybase::swap(_Right);
+        _STD swap(_Buf, _Right._Buf);
+    }
+
+    void swap(basic_spanstream& _Right) {
+        _Mybase::swap(_Right);
+        _STD swap(_Buf, _Right._Buf);
+    }
+
+    // N4892 [spanstream.members], members

```

Comment:
Hyper-nitpick, the previous comments (and the Standard section name) said "member functions".